src/sync/sync_session.cpp

```diff
@@ -637,7 +641,7 @@ void SyncSession::handle_error(SyncError error)
             break;
         case NextStateAfterError::inactive: {
             std::unique_lock<std::mutex> lock(m_state_mutex);
```

Comment:
Instead of passing a boolean down the stack, I think it would be cleaner to do something like the following before changing states here:
```
if (error.is_client_reset_requested()) {
    cancel_pending_waits(lock, make_error_code(sync::ProtocolError::bad_client_file));
}
```

Reply:
`cancel_pending_waits()` will unlock so I cannot change the state.

Reply:
Yes, that's a big simplification.